### ISSUE TITLE
Define semantics of `num_updates` for scikit-learn

### ISSUE THREAD

**Issue by srcansiz** [fedbiomed/fedbiomed#371]
In GitLab by @sharkovsky on Sep 23, 2022, 08:47

Created from #369 

The semantics of `num_updates` may be confusing for scikit-learn.
We have started the following discussion on Discord

> What should num_updates mean in this case, considering that the minimum update we can make with sklearn is 1 epoch?
> - if we take it to mean the same as num epochs, then we'll have this weird situation where you ask for 500 updates in a TorchTrainingPlan, and for 2 updates in the SKLearnTrainingPlan, and they'll "see" the same number of samples
> - if we take it to mean the same as the number of samples seen, then we will lie to the researcher. For example, in a dataset with 300 samples, if the researcher asks for num_updates=500 then we'll either do one epoch (i.e. 300 samples seen) or two epochs (i.e. 600 samples seen)

@ybouilla has confirmed that this is a scikit-learn limitation. 

@pandrey-fr has suggested the following nifty solution:
> This is indeed a limitation with scikit-learn, that occurs in most cases since most scikit-learn models run are designed to handle the entire training procedure based on a single fit call.
> This can however be avoided the specific case of SgdClassifier / SgdRegressor classes, because it is possible to set a constant arbitrary learning rate, feed the partial_fit method with a single data sample at a time, and reset its weights after each call. This way, it is possible to compute sample-wise, and therefore batch-wise, gradients.
> This is what we do in declearn, so that the same optimizer code (and plug-in system) can be used with these models as with torch and tensorflow.
> 
> Note that another way could be to re-implement scikit-learn models in another framework (including mere numpy) to avoid having to resort to such tricks. 

We need to make a decision together on which approach to take.
